EarlGrey/Event/GREYSyntheticEvents.m

```diff
@@ -112,6 +113,41 @@ + (BOOL)rotateDeviceToOrientation:(UIDeviceOrientation)deviceOrientation
   return YES;
 }
 
++ (BOOL)shakeDeviceWithError:(NSError *__strong *)errorOrNil {
+  GREYFatalAssertMainThread();
+  
+  NSError *error;
+  BOOL success = [[GREYUIThreadExecutor sharedInstance] executeSyncWithTimeout:kRotationTimeout
+                                                                         block:^{
+    // Keep previous accelerometer events enabled value and force it to YES so that the shake
+    // motion is passed to the application.
+    UIApplication *application = [UIApplication sharedApplication];
+    BOOL prevValue =
+        [[[application _motionEvent] valueForKey:@"_motionAccelerometer"]
+            accelerometerEventsEnabled];
+    [[[[UIApplication sharedApplication] _motionEvent] valueForKey:@"_motionAccelerometer"]
+        setAccelerometerEventsEnabled:YES];
+    
+    // This behaves exactly in the same manner that UIApplication handles the simulator
+    // "Shake Gesture" menu command.
+    [[UIApplication sharedApplication] _sendMotionBegan:UIEventSubtypeMotionShake];
+    [[UIApplication sharedApplication] _sendMotionEnded:UIEventSubtypeMotionShake];
+    
+    [[[[UIApplication sharedApplication] _motionEvent] valueForKey:@"_motionAccelerometer"] setAccelerometerEventsEnabled:prevValue];
```

Comment:
use variable which stores return val of `[[UIApplication sharedApplication] _motionEvent] valueForKey:@"_motionAccelerometer"]`